bsp/nuclei/tools/sdk_dist.py

```diff
@@ -5,11 +5,10 @@
 sys.path.append(os.path.join(os.path.dirname(cwd_path), 'rt-thread', 'tools'))
 
 # BSP dist function
-def dist_do_building(BSP_ROOT):
+def dist_do_building(BSP_ROOT, dist_dir):
```

Comment:
Please not modify other files.

Reply:
If I don't modify this function, it will show error.

Reply:
I also noticed that in file mkdist.py, there are two different function call for dist_handle function, which one is correct?

1. https://github.com/RT-Thread/rt-thread/blob/master/tools/mkdist.py#L217-L221
2. https://github.com/RT-Thread/rt-thread/blob/master/tools/mkdist.py#L341-L343

Reply:
I see there are three different way of distribution source code of RT-Thread. But only `scons --dist` is documented, others like `scons --dist-strip` and `scons --dist-ide` are not documented, and it also seems that these two ways generated folders and structures are different.

I only tested `scons --dist` which works for me.

And also for the ide project generation, I created a empty eclipse ide project using nuclei studio, and want to use `scons --target=eclipse` to generate it, it will show error in parsing the project templates.

Reply:
@fanghuaqi `scons --target=eclipse` command only supports stm32 series bsp now, as a user of bsp, using `scons --dist` command is enough, `scons --dist-ide` command is used for RT-Thread Studio to generate project base on bsp already existing, you don't need this command.

Reply:
> ?

@SummerLife Could you also check this one? `scons --dist-strip`? And if I want to support `scons --target=eclipse` for Nuclei Studio which is also eclipse-based IDE, could you also support it?

Thanks
Huaqi

Reply:
This [link](https://github.com/RT-Thread/rt-thread/blob/ce833715042dfae615f3bc5ca22b8d294b7cade0/tools/mkdist.py#L340) is the only one need notice, i‘m’ not sure i have enough time to support a new bsp series for eclipse IDE.

Reply:
Maybe the sdk_dist.py in Nuclei need update indeed, you can test the `scons --dist` on that bsp, if it can't works, you can summit a new pr to repair it.

Reply:
@SummerLife , hi, for `scons --dist-strip` and `scons --dist` which one is now recommended, this two commands will show different behavious, and the `dist_handle` function used have different parameter number, if I want to implement it in bsp/nuclei/tools, then it will be hard to select which one I should implement.

Reply:
> 
> 
> Maybe the sdk_dist.py in Nuclei need update indeed, you can test the `scons --dist` on that bsp, if it can't works, you can summit a new pr to repair it.

Hi, I tested on this PR, the `scons --dist` works well, but `scons --dist-strip` doesn't.
![图片](https://user-images.githubusercontent.com/1538922/84217527-fb22d300-aafe-11ea-9408-d131c83439e8.png)

![图片](https://user-images.githubusercontent.com/1538922/84217542-05dd6800-aaff-11ea-9644-d6a29c4996cf.png)

Reply:
i think the `scons --dist` command  works well is enough now, i rarely use `scons --dist-strip` comand.

Reply:
Hi @SummerLife , yes, it works, but what bothers me is that the two different implementations are not well documented, and which one is recommended for RT-Thread users, as follows:

* code in `def MkDist_Strip(program, BSP_ROOT, RTT_ROOT, Env):`

```python
    # do bsp special dist handle
    if 'dist_handle' in Env:
        print("=> start dist handle")
        dist_handle = Env['dist_handle']
        dist_handle(BSP_ROOT)
```

* code in `def MkDist(program, BSP_ROOT, RTT_ROOT, Env, rttide = None):`

```python
    # do bsp special dist handle
    if 'dist_handle' in Env:
        print("=> start dist handle")
        dist_handle = Env['dist_handle']
        dist_handle(BSP_ROOT, dist_dir)
```

Reply:
You can just use the commamd `scons --dist` and function `MkDist`, no need to find what's the different between `scons --dist` and `scons --dist-strip`.

Reply:
OK, thanks. I want to suggest that if `scons --dist-strip` is not working well and not documented, maybe RT-Thread team can remove that option in the help message of `scons --help`